index.js

```diff
@@ -99,6 +99,23 @@ class Ora {
 	}
 }
 
+function promise(action, options) {
+	if (typeof action.then === 'function') {
```

Comment:
This should be reversed as a typecheck for `throw new Error('Parameter `action` must be a Promise');`. No need to wrap the main code here in an if-statement.